optaplanner-examples/src/test/java/org/optaplanner/examples/conferencescheduling/optional/score/ConferenceSchedulingConstraintProviderTest.java

```diff
@@ -0,0 +1,919 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.examples.conferencescheduling.optional.score;
+
+import static java.util.Collections.emptySet;
+import static java.util.Collections.singleton;
+import static java.util.Collections.singletonList;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Arrays;
+import java.util.HashSet;
+
+import org.junit.jupiter.api.Test;
+import org.optaplanner.examples.conferencescheduling.domain.ConferenceConstraintConfiguration;
+import org.optaplanner.examples.conferencescheduling.domain.ConferenceSolution;
+import org.optaplanner.examples.conferencescheduling.domain.Room;
+import org.optaplanner.examples.conferencescheduling.domain.Speaker;
+import org.optaplanner.examples.conferencescheduling.domain.Talk;
+import org.optaplanner.examples.conferencescheduling.domain.Timeslot;
+import org.optaplanner.test.api.score.stream.ConstraintVerifier;
+
+public class ConferenceSchedulingConstraintProviderTest {
+
+    private final ConstraintVerifier<ConferenceSchedulingConstraintProvider, ConferenceSolution> constraintVerifier =
+            ConstraintVerifier.build(new ConferenceSchedulingConstraintProvider(), ConferenceSolution.class,
+                    Talk.class);
+
+    private static final LocalDateTime START = LocalDateTime.ofInstant(Instant.EPOCH, ZoneId.systemDefault());
+
+    private static final Timeslot TIMESLOT1 = new Timeslot(1)
+            .withStartDateTime(START)
+            .withEndDateTime(START.plusHours(1))
+            .withTagSet(singleton("a"));
```

Comment:
A variable name like TIMESLOT1 etc makes the tests harder to understand.
Now I first have to build a mental mapping of timeslot 1, 2, 3 and 4 on how they overlap and they gaps, and then read the constraint test to validate if the assertions are asserting the right thing.

Also, a conference starting at midnight 1 jan 1970 isn't a realistic test I'd argue.

Proposal: instead of TIMESLOT_1 can we make it MONDAY_9_TO_10 and use "1-Feb 2000 9:00" as a value?
If a constraint test then checks if a speaker assigned to talks on MONDAY_9_TO_10 and MONDAY_9_TO_11 overlap, that's much easier to read.